lib/jxl/encode.cc

```diff
@@ -1588,6 +1634,53 @@ JxlEncoderStatus JxlEncoderAddJPEGFrame(
   return JXL_ENC_SUCCESS;
 }
 
+static bool CanDoFastLossless(const JxlEncoderFrameSettings* frame_settings,
+                              const JxlPixelFormat* pixel_format,
+                              bool has_alpha) {
+  if (!frame_settings->values.lossless) {
+    return false;
+  }
+  if (frame_settings->values.frame_index_box) {
+    return false;
+  }
+  if (frame_settings->values.cparams.speed_tier != jxl::SpeedTier::kLightning) {
+    return false;
```

Comment:
Not for this PR, but I would eventually use fast lossless at both e1 and e2 (with different fast effort settings, e.g. no palette check and cheaper histogram sampling at e1), and then make e3 do something that also tries the fast path since it can very well give better results for non-photo with big run lengths.

I'm not sure if we still need the current e2 except as a fallback option for cases not covered by the fast path.